packages/react-server/src/ReactFlightServer.js

```diff
@@ -708,6 +708,7 @@ function flushCompletedChunks(request: Request): void {
         break;
       }
     }
+    moduleChunks.splice(0, i);
```

Comment:
This is a bug fix that this test revealed. This causes the connection to close to early because we keep emitting module chunks over and over since they don't get deleted.